src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -10325,7 +10372,7 @@ and TcAndBuildFixedExpr (cenv: cenv) env (overallPatTy, fixedExpr, overallExprTy
                     | _ -> false
             | _ -> false
         if not okByRef then
-            error(Error(FSComp.SR.tcFixedNotAllowed(), mBinding))
+            checkLanguageFeatureError g.langVersion LanguageFeature.ExtendedFixedBindings mBinding
```

Comment:
I'm slightly confused here, why was the error changed in this case?

Reply:
It's because this PR makes all byref expressions work with fixed bindings, not just a subset. Take this code for example (from [this discussion](https://github.com/fsharp/fslang-design/discussions/421#discussioncomment-6511307)):

```fsharp
type MyType() =
    let mutable value = 42
    
    member this.Pin () =
        use ptr = fixed &value
        printfn "%i" (NativePtr.read ptr)
        ()
```

F# 7 rejects this with FS3207. The new version allows this construct (it passes through this branch to do so). If the feature is turned off, it now tells you that in order to use fixed in this way you must turn the feature on / upgrade the compiler.

Perhaps it could be rewritten to be a little clearer -- I realize now that if the feature flag is on, it doesn't need to check for `okByRef` at all.